File: cpp/718.cpp
int findLength(int* A, int ASize, int* B, int BSize) {
    int dp[ASize+1][BSize+1];
    memset( dp, 0, (ASize+1)*(BSize+1)*sizeof(int) );
    int max =0;
    for(int i=1;i<=ASize;i++)
        for(int j=1;j<=BSize;j++)
        { 
            if(A[i-1]==B[j-1]){
                dp[i][j]=dp[i-1][j-1] + 1;
                if(dp[i][j]>max)
                    max=dp[i][j];       
            }  
        }
    return max;
}

File: ruby/718.rb
# @param {Integer[]} a
# @param {Integer[]} b
# @return {Integer}
def find_length(a, b)
    m = a.size
    n = b.size
    dp = m.times.map{ [0] * n }
    for i in 0..m-1
        for j in 0..n-1
            if a[i] == b[j]
                if i ==0 || j==0
                    dp[i][j] = 1
                else
                    dp[i][j] = dp[i-1][j-1] + 1
                end
            else
                dp[i][j] = 0
            end
        end
    end
    # p dp
    return dp.flatten.max
end
# a = [0,0,0,0,1]
# b = [1,0,0,0,0]

# p find_length(a, b)